### ISSUE TITLE
DalResults does not handle 404 (or any error) sanely

### ISSUE THREAD

**Issue by mtauraso** [astropy/pyvo#625]
Any calls to DalResult.from_result_url resulting in an error condition (404, 5xx etc) will proceed to trying to parse the response as a votable.

For a user calling a higher-level function like DataLinkResults.from_result_url the resulting error is a parse error deep inside votable and distracts from the actual problem (the service doesn't have what they're looking for).

I've filed an example of this behavior [here](url) TODO PUT IN URL

The issue is [here](https://github.com/astropy/pyvo/blob/2bf4e3f1ffea04fa168713a715b0058efed77d6e/pyvo/dal/query.py#L301) where either `from_result_url` or `_from_result_url` should check `.status` on the `HTTPResponse` object returned from [`session.get`](https://github.com/astropy/pyvo/blob/2bf4e3f1ffea04fa168713a715b0058efed77d6e/pyvo/auth/authsession.py#L60)

I'm happy to submit a PR if there is support for this fix.

**Comment by bsipocz** [comment]
Yes, pyvo specific dev docs are coming. Until then the jist is: Keeping the commit history clean would be appreciated, and rebased when/if you need to refresh the base branch rather than github resolve and add a merge commits. Also, please refrain from running auto reformatting anything or add changes unrelated to the particular fix, if you run into any other bugs or stuff it's preferred to fix them in a separate PR.

And we have two types of tests, locally mocked ones, but also some functionality has tests that reaches out to the remote servers. You can run the latter lot by using the `-R` pytest option.
